pkg/functions/openfaas/driver.go

```diff
@@ -94,6 +122,25 @@ func (d *ofDriver) Create(f *functions.Function, exec *functions.Exec) error {
 		}
 		return errors.Wrapf(err, "Error performing POST request, status: %v", res.StatusCode)
 	}
+
+	// make sure the function has started
+	for i := 0; i < 10; i++ {
+		deployment, err := d.deployments.Get(getID(f.ID), v1.GetOptions{})
+		if err != nil {
+			log.Errorf("%+s", errors.Wrapf(err, "failed to read function deployment status: '%s'", getID(f.ID)))
+
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		if deployment.Status.AvailableReplicas > 0 {
+			return nil
+		}
+
+		time.Sleep(1 * time.Second)
+	}
+
+	return errors.Errorf("Failed to start function deployment: '%s'", getID(f.ID))
```

Comment:
I assume this will put the function in error state?

Reply:
Correct